File: Data Structures & Algorithms/decode-ways/submission-8.cpp
class Solution {
    int result=0;
public:
    void decode(string s,int index) {
        if(s[index]!='0') {
            result+=1;
            int i=index;
            while(i<s.length()-1) {
                if(s[i]=='2') {
                    if(s[i+1]=='1' ||s[i+1]=='2' || s[i+1]=='3' || s[i+1]=='4' || s[i+1]=='5' || s[i+1]=='6') {
                        decode(s,i+2);
                        //i+=1;
                    }
                    i+=1;
                }
                else if(s[i]=='1' && s[i+1]!='0') {
                    
                    decode(s,i+2);
                    i+=1;
                }
                else i+=1;
            }
        }
        
    }

    int numDecodings(string s) {
        if(s[0]=='0')return result;
        for(int i=0;i<s.length()-1;i++) {
            if(s[i]=='0' && s[i+1]=='0')return 0;
            if((s[i]!='1'&&s[i]!='2')&&(s[i+1]=='0'))return 0;
        }
        decode(s,0);
        return result;
       
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-8.py
class HashList:
    def __init__(self,n):
        self.li=[[] for _ in range(n)]
        self.n=n
    
    def insert(self,val):
        index=val%self.n
        if val not in self.li[index]:
            self.li[index].append(val)
    
    def isPresent(self,val):
        index=val%self.n
        if val in self.li[index]:
            return True
        return False

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash_list=HashList(10)
        for val in nums:
            if hash_list.isPresent(val)==True:
                return True
            hash_list.insert(val)
        
        return False